coding.go

```diff
@@ -34,64 +34,63 @@ func (nb _PBNode__NodeBuilder) DecodeDagProto(r io.Reader) error {
 	if err := pbn.Unmarshal(encoded); err != nil {
 		return fmt.Errorf("unmarshal failed. %v", err)
 	}
-	pbLinks := make([]PBLink, 0, len(pbn.Links))
-	for _, link := range pbn.Links {
-		hash, err := cid.Cast(link.GetHash())
+	return fluent.Recover(func() {
+		fb := fluent.WrapAssembler(nb)
+		fb.CreateMap(0, func(fmb fluent.MapAssembler) {
+			fmb.AssembleEntry("Links").CreateList(len(pbn.Links), func(flb fluent.ListAssembler) {
+				for _, link := range pbn.Links {
+					hash, err := cid.Cast(link.GetHash())
 
-		if err != nil {
-			return fmt.Errorf("unmarshal failed. %v", err)
-		}
-		pbLinks = append(pbLinks, PBLink{
-			d: PBLink__Content{
-				Hash: MaybeLink{
-					Maybe: schema.Maybe_Value,
-					Value: Link{cidlink.Link{Cid: hash}},
-				},
-				Name: MaybeString{
-					Maybe: schema.Maybe_Value,
-					Value: String{link.GetName()},
-				},
-				Tsize: MaybeInt{
-					Maybe: schema.Maybe_Value,
-					Value: Int{int(link.GetTsize())},
-				},
-			},
+					if err != nil {
+						panic(fluent.Error{Err: fmt.Errorf("unmarshal failed. %v", err)})
+					}
+					flb.AssembleValue().CreateMap(0, func(fmb fluent.MapAssembler) {
+						fmb.AssembleEntry("Hash").AssignLink(cidlink.Link{Cid: hash})
+						fmb.AssembleEntry("Name").AssignString(link.GetName())
```

Comment:
I'm moderately confused what the protobuf library is doing here -- we had to hand the protobuf lib a pointer to the name string elsewhere, because it's option in protobuf's mind, right?  But here it's returning a string (no pointer).  Does this panic if the string is missing?  Or silently coerce it to an empty string?

Reply:
yea it silently coerces it tp empty string. Which is weird behavior except -- that's exactly what go-merkledag uses in in IPFS. My basic approach is follow as close as possible so as to not break things.

Reply:
Okay, cool.  I appreciate that, thanks for talking through it with us so we have it also as input to the other standardization quests @rvagg mentioned :)